lib/Support/Pipeline.cpp

```diff
@@ -42,8 +42,10 @@ void MappingTraits<offloadtest::Pipeline>::mapping(IO &I,
       if (!R.ExpectedPtr)
         I.setError(Twine("Reference buffer ") + R.Expected + " not found!");
       if (R.Rule == offloadtest::Rule::BufferFuzzy) {
-        if (R.ActualPtr->Format != offloadtest::DataFormat::Float32 ||
-            R.ExpectedPtr->Format != offloadtest::DataFormat::Float32)
+        if ((R.ActualPtr->Format != offloadtest::DataFormat::Float32 ||
```

Comment:
I think it would be good to write a function like 'isFloatingPointFormat' which checks if the format is Float16 or Float32. This would be useful when we need to add Float64 to this check.